bpe.go

```diff
@@ -173,6 +190,101 @@ func ReadModel(reader io.Reader) (*Model, error) {
 		return &Model{}, err
 	}
 	specials, err := binaryToSpecialTokens(specialTokensBuf)
+	if err != nil {
+		return model, err
+	}
 	model.specialTokens = specials
 	return model, err
 }
+
+// IDToToken returns string token corresponding to the given token id.
+// If replaceSpace is true, special space token that is used for marking starts of words
+// will be replaced with space.
+func (m Model) IDToToken(id TokenID, replaceSpace bool) (string, error) {
+	if _, ok := m.recipe[id]; !ok {
+		if id == TokenID(m.specialTokens.unk) {
```

Comment:
``go
switch id {
case TokenID(m.specialTokens.unk):
    ...
default:
    logrus.Errorf("%d: token id is impossible", id)
}
```

Note: "impossible" instead of OOV because you already have `<UNK>`